### PR TITLE
Generate CSVs for homepage link status and interaction link status

### PR DESCRIPTION
**Pull request by issyl0**
- This generates CSVs of link check statistics when people access `/check_homepage_links_status.csv` and `/check_links_status.csv`, so we can do reporting into the numbers of broken links etc.
- Trello: https://trello.com/c/BCMg9RTo/426-get-the-data-for-broken-links-2

(Paired with @klssmith.)

### CODE REVIEW

File: app/presenters/link_check_csv_presenter.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      LocalAuthority.group(:status).count.each do |key, value|
+        csv << [key || "nil", value]
+      end
+    end
+  end
+
+  def self.links_status_csv
+    CSV.generate do |csv|
+      csv << %w(status count)
+      Link.group(:status).count.each do |key, value|
```
Comment (jennyd) [inline app/presenters/link_check_csv_presenter.rb]: Could we scope this to only include links for services which are enabled? All other links aren't checked at all, so it's harder to work out how many links should be checked but haven't been yet for some reason (and so still have `nil` as their status) if we include all the links we'll never check.

Comment (sihugh) [inline app/presenters/link_check_csv_presenter.rb]: I agree although it does slightly beg the question why we hold on to old links for services that we don't care about and don't check.  Should we either check them or get rid of them?  (That's not really to do with this PR though, sorry)

Comment (jennyd) [inline app/presenters/link_check_csv_presenter.rb]: We may enable other services in future, and having the old links in the database already would give us a starting point for those. The checking takes a while, though, and would take much longer if we included all the disabled service links - we don't need quality data until we want to start using them.

File: spec/controllers/links_controller_spec.rb [hunk 2]
```diff
@@ -2,26 +2,38 @@
...
       expect(response).to have_http_status(302)
       expect(flash[:danger]).not_to be_present
     end
   end
+
+  describe 'GET homepage_links_status_csv' do
+    it "retrieves HTTP success" do
+      get :homepage_links_status_csv
+      expect(response).to have_http_status(:success)
+    end
```
Comment (sihugh) [inline spec/controllers/links_controller_spec.rb:31]: We have tests for generating the CSV and ensuring that we can get a successful response from the route, but nothing that knits the two together.  Maybe we could have a feature spec that ensures that we serve a CSV when you hit this endpoint and it has the expected columns?

File: app/presenters/link_check_csv_presenter.rb (hunk 1 above)
Comment (sihugh) [inline app/presenters/link_check_csv_presenter.rb]: I understand, it just makes me feel slightly icky having it hanging around uncared for.  I'll worry more about it in the Autumn. :-)

Comment (issyl0) [inline app/presenters/link_check_csv_presenter.rb]: Updated.

**Comment by sihugh** [comment]
🎫

**Pull request merged**